google/api_core/grpc_helpers.py

```diff
@@ -113,6 +116,10 @@ def __next__(self) -> P:
                 result = self._stored_first_result
                 del self._stored_first_result
                 return result
+            elif hasattr(self, "_stored_first_error"):
+                exc = self._stored_first_error
+                del self._stored_first_error
+                raise exc
```

Comment:
I'm considering whether it's worth refactoring both these variables, so we don't have to do two `hasattr` lookups for each stream result. For now I'm just following the existing code though